src/core/dfn-index.js

```diff
@@ -0,0 +1,214 @@
+  if (!index) {
+    return;
+  }
+
+  const styleEl = document.createElement("style");
+  styleEl.textContent = await loadStyle();
+  document.head.appendChild(styleEl);
+
+  index.classList.add("appendix");
+  if (!index.querySelector("h2")) {
+    index.prepend(html`<h2>Index</h2>`);
+  }
+
+  const toCiteDetails = citeDetailsConverter(conf);
+
+  const externalTermIndex = html`<section id="index-defined-elsewhere">
+    <h3>Terms defined by reference</h3>
+    ${createExternalTermIndex(toCiteDetails)}
+  </section>`;
+  index.append(externalTermIndex);
+
+  sub("beforesave", cleanup);
+}
+
+/**
+ * @param {ReturnType<typeof citeDetailsConverter>} toCiteDetails
+ */
+function createExternalTermIndex(toCiteDetails) {
+  const data = collectExternalTerms(toCiteDetails);
+  const dataSortedBySpec = [...data.entries()].sort((a, b) =>
+    a[0].localeCompare(b[0])
+  );
+  return html`<ul class="index">
+    ${dataSortedBySpec.map(
+      ([spec, entries]) => html`<li data-spec="${spec}">
+        ${renderInlineCitation(spec)} defines the following:
+        <ul>
+          ${entries
+            .sort((a, b) => a.term.localeCompare(b.term))
+            .map(renderExternalTermEntry)}
+        </ul>
+      </li>`
+    )}
+  </ul>`;
+}
+
+/**
+ * @param {ReturnType<typeof citeDetailsConverter>} toCiteDetails
+ */
+function collectExternalTerms(toCiteDetails) {
+  /** @type {Set<string>} */
+  const uniqueReferences = new Set();
+  /** @type {Map<string, Entry[]>} spec => entry[] */
+  const data = new Map();
+
+  /** @type {NodeListOf<HTMLAnchorElement>} */
+  const elements = document.querySelectorAll(`a[data-cite]`);
+  for (const elem of elements) {
+    if (!elem.dataset.cite) {
+      continue;
+    }
+    const uniqueID = elem.href;
+    if (uniqueReferences.has(uniqueID)) {
+      continue;
+    }
+
+    const { type, linkFor } = elem.dataset;
+    const term = getTermFromElement(elem);
+    if (!term) {
+      continue; // <a data-cite="SPEC"></a>
+    }
+    const spec = toCiteDetails(elem).key.toUpperCase();
+
+    const entriesBySpec = data.get(spec) || data.set(spec, []).get(spec);
+    entriesBySpec.push({ term, type, linkFor, elem });
+    uniqueReferences.add(uniqueID);
+  }
+
+  return data;
+}
+
+/**
+ * @param {Entry} entry
+ * @returns {HTMLLIElement}
+ */
+function renderExternalTermEntry(entry) {
+  const { elem } = entry;
+  const text = getTermText(entry);
+  const el = html`<li>
+    <span class="index-term" data-href="${elem.href}">${{ html: text }}</span>
+  </li>`;
+  addId(el.querySelector("span"), "index-term");
+  return el;
+}
+
+// Terms of these _types_ are wrapped in `<code>`.
+const CODE_TYPES = new Set([
+  "attribute",
+  "callback",
+  "dict-member",
+  "dictionary",
+  "element-attr",
+  "element",
+  "enum-value",
+  "enum",
+  "exception",
+  "extended-attribute",
+  "interface",
+  "method",
+  "typedef",
+]);
+
+// Terms of these _types_ are suffixed with their type info.
+const TYPED_TYPES = new Map([
```

Comment:
Maybe it could be we only have the special cases here... if it's not in the TYPED_TYPES, then just use what is given?  🤔 doesn't really matter too much tho.

Reply:
It gets noisy otherwise, so limited what gets type info. For example, "creating an event dfn", "foo enum-value".
Also, it's used to map human-friendly terms, like { x, element-attr } becomes "x attribute (for y element)"